projects/schematics/src/migrations/4_0/constructor-deprecations/data/cart-item-list.component.migration.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ACTIVE_CART_SERVICE,
+  CART_ITEM_LIST_COMPONENT,
+  FEATURE_CONFIG_SERVICE,
+  MULTI_CART_SERVICE,
+  SELECTIVE_CART_SERVICE,
+  SPARTACUS_CORE,
+  SPARTACUS_STOREFRONTLIB,
+  USER_ID_SERVICE,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const CART_ITEM_LIST_COMPONENT_MIGRATIONS: ConstructorDeprecation[] = [
+  // projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts
+  {
+    class: CART_ITEM_LIST_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    deprecatedParams: [
+      { className: ACTIVE_CART_SERVICE, importPath: SPARTACUS_CORE },
+      { className: SELECTIVE_CART_SERVICE, importPath: SPARTACUS_CORE },
+    ],
+    addParams: [
+      { className: USER_ID_SERVICE, importPath: SPARTACUS_CORE },
+      { className: MULTI_CART_SERVICE, importPath: SPARTACUS_CORE },
+    ],
+  },
+  {
+    class: CART_ITEM_LIST_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    deprecatedParams: [
+      { className: ACTIVE_CART_SERVICE, importPath: SPARTACUS_CORE },
+      { className: SELECTIVE_CART_SERVICE, importPath: SPARTACUS_CORE },
+      { className: FEATURE_CONFIG_SERVICE, importPath: SPARTACUS_CORE },
+    ],
+    removeParams: [
+      { className: FEATURE_CONFIG_SERVICE, importPath: SPARTACUS_CORE },
+    ],
+    addParams: [
+      { className: USER_ID_SERVICE, importPath: SPARTACUS_CORE },
+      { className: MULTI_CART_SERVICE, importPath: SPARTACUS_CORE },
+    ],
+  },
```

Comment:
To be complete we should also add a case when the customer provided all 5 services and we need to only remove the feature config service.